src/dune_engine/source_tree.ml

```diff
@@ -491,8 +493,24 @@ end = struct
               "Note: You can use \"dune upgrade\" to convert your project to \
                dune."
           ]
+      else if Dune_project.dune_version project < (3, 0) then
+        if String.Set.mem files Dune_file.dune_fname then
+          Some Dune_file.dune_fname
+        else
+          None
```

Comment:
Might be nice to print a warning if the `dune-file` exists, saying that it is only supported sincee 3.0